sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/DefaultService.java

```diff
@@ -72,17 +73,16 @@ public DefaultService(DefaultScheduler.Builder schedulerBuilder) throws Exceptio
     }
 
     public static Boolean serviceSpecRequestsGpuResources(ServiceSpec serviceSpec) {
-        Collection<PodSpec> pods = serviceSpec.getPods();
-        for (PodSpec pod : pods) {
-            for (TaskSpec taskSpec : pod.getTasks()) {
-                for (ResourceSpec resourceSpec : taskSpec.getResourceSet().getResources()) {
-                    if (resourceSpec.getName().equals("gpus") && resourceSpec.getValue().getScalar().getValue() >= 1) {
-                        return true;
-                    }
-                }
-            }
-        }
-        return false;
+        boolean usesGpus = serviceSpec.getPods().stream()
+                .flatMap(podSpec -> podSpec.getTasks().stream())
+                .flatMap(taskSpec -> taskSpec.getResourceSet().getResources().stream())
+                .collect(Collectors.toList()).stream()
+                .filter(resourceSpec -> resourceSpec.getName().equals("gpus"))
+                .filter(resourceSpec -> resourceSpec.getValue().getScalar().getValue() >= 1)
+                .count() >= 1;
+        // control automatic opt-in to scarce resources (GPUs) here. If the framework specifies GPU resources >= 1
+        // then we opt-in to scarce resource, otherwise follow the default policy (which as of 8/3/17 was to opt-out)
+        return usesGpus || DcosConstants.DEFAULT_GPU_POLICY;
```

Comment:
Since `DEFAULT_GPU_POLICY` is `false` (and is `static final`), does this not mean that this conditional or statement is not required? Or is there some way that `DEFAULT_GPU_POLICY` could change?